src/equations/compressible_euler_1d.jl

```diff
@@ -853,23 +857,23 @@ Compactly summarized:
     v_roe_mag = v_roe^2
 
     H_roe = (sqrt_rho_ll * H_ll + sqrt_rho_rr * H_rr) * inv_sum_sqrt_rho
-    c_roe = sqrt((equations.gamma - 1) * (H_roe - 0.5 * v_roe_mag))
+    c_roe = sqrt((equations.gamma - 1) * (H_roe - 0.5f0 * v_roe_mag))
 
     # Compute convenience constant for positivity preservation, see
     # https://doi.org/10.1016/0021-9991(91)90211-3
-    beta = sqrt(0.5 * (equations.gamma - 1) / equations.gamma)
+    beta = sqrt(0.5f0 * (equations.gamma - 1) / equations.gamma)
 
     # Estimate the edges of the Riemann fan (with positivity conservation)
-    SsL = min(v_roe - c_roe, v_ll - beta * c_ll, zero(v_roe))
-    SsR = max(v_roe + c_roe, v_rr + beta * c_rr, zero(v_roe))
+    SsL = min(v_roe - c_roe, v_ll - beta * c_ll, 0)
```

Comment:
Is this guaranteed to be type stable?

Reply:
It's not guaranteed from the docstrings but from the implementations:

```julia
max(x::Real, y::Real) = max(promote(x,y)...)
min(x::Real, y::Real) = min(promote(x,y)...)
```

Reply:
Should we then just keep it as it was before?

Reply:
See below